### DISCUSSION TITLE
combine multiple tables

### DISCUSSION THREAD

**Discussion by 88arvin** [jsvine/pdfplumber#786] [Q&A] [answered]
I have a pdf with 10 pages, and some pages have one table and some pages have multiple tables. When I am using "extract table()", it is extracting the largest table on each page but I want to extract all the tables and append them into a single dataframe.
How to append every table into a single dataframe using "extract_tables()".

**Comment by jsvine** [comment]
Hi @88arvin, thanks for your interest in this library. If I'm understanding your question correctly, the answer has less to do with `pdfplumber` and is more generic to Python (or, since you used the phrase "dataframe", `pandas`). Here's what I'd suggest:

```python
core = []  # Start with a blank list
for page in pdf.pages:  # Iterate over the pages
  core += page.extract_tables()  # In Python, += extends a list
df = pd.DataFrame(core)  # Assuming that, by "dataframe," you mean a pandas DataFrame
```

**Comment by 88arvin** [comment, marked as answer]
Thank you for responding so quickly. I have tried your suggested code; however, the data is still not coming in the form of a dataframe.
[https://postimg.cc/3W86SH0d](url)
If I use extract_table(), then I am able to iterate through all the pages and able to append the data into a single dataframe, but only able to extract the largest table from each page.

**Reply by jsvine** [reply]
Ah, I see what you mean. Try this:

```python
core = []  # Start with a blank list
for page in pdf.pages:  # Iterate over the pages
  for table in page.extract_tables():
    core +=  table # In Python, += extends a list
df = pd.DataFrame(core)  # Assuming that, by "dataframe," you mean a pandas DataFrame
```

**Reply by 88arvin** [reply]
Oh  yes! This worked well, but it extracted not only the tables but also all the other data.

**Reply by 88arvin** [reply]
How to only extract tables?
